src/mlpack/methods/rbm/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# Define the files we need to compile
+# Anything not in this list will not be compiled into mlpack.
+set(SOURCES
+  binary_layer.hpp
+  binary_layer_impl.hpp
+  binary_rbm.hpp 
+  rbm.hpp
```

Comment:
In my opinion it is better to move these files back to `mlpack/methods/ann` since other neural networks are located there.